proposals/default_optional_arguments/proposal.txt

```diff
@@ -141,8 +142,36 @@ A use case of this behavior of the function present() could be:
       ...
     end function foo
 
+5.2. intent(out)
+
+There are several different possible behaviors for intent(out)
+variables. One option is to disallow default values on optional
+intent(out) variables. However, there are use cases where it is
+desirable to return the value of a variable if a caller requests it,
+without requiring the added logic of present() and "temporary"
+variables described above.
+
+Alternatively, we might treat intent(in) and intent(out) variables
+identically, so that the following would be valid:
+
+    real function foo(a)
+      real, intent(out), optional :: a = 0
+    end function foo
+
+The benefit is that, just like proposed for intent(in) variables,
+intent(out) variables would be usable regardless of whether the value
+is actually passed back to the caller, removing the need for present()
+and "temporary" variables. The complication is that a present
+intent(out) variable is initially undefined, meaning the logic of
+present() is still necessary in this case. Furthermore, the behavior
+suggested above where present() = .true. always for variables with a
+specified default would not be desired for intent(out) variables.
+
+A third option is that optional intent(out) variables with a specified
+default are always initialized to that default, even when present,
+rather than leaving them undefined.
 
```

Comment:
The scenario `intent(inout)` (or no specification of `intent()`) is missing. Its behavior would probably depend on the behaviors of `intent(in)` and `intent(out)`, e.g., as proposed by @everythingfunctional (which might not be in agreement with all different proposed options for `intent(out)`):

"If it is intent(inout) and the caller provides an actual argument, it wil get the value of the actual argument, AND the returned value is equal to the actual value if it is not modified, or the assigned value otherwise. (Use case: @everythingfunctional 's example)."

Should it be mentioned?